lib/game.rb

```diff
@@ -27,16 +29,31 @@ def take_turn(board, position, token)
         end
     end
 
-    def game_over?(board)
+    def game_over?(board, player)
         if @game_result.draw?(board)
-            report_game_result
+            report_game_result(display_draw_message, player=nil)
             puts @presenter.display_board(board.moves)
             return true
+        elsif @game_result.winner?(board)
+            winner = display_winner_message(player)
+            report_game_result(winner, player)
+            puts @presenter.display_board(board.moves)
+            return true
+        else 
+            false
         end
     end
 
-    def report_game_result
-        puts "It's a draw! Game over!"
+    def display_draw_message
+        @prompt.draw_message
+    end
+
+    def display_winner_message(player)
+        @prompt.winner_message(player)
+    end
+
+    def report_game_result(result, player)
```

Comment:
What's the point of this if it's just returning result? Is it a WIP?

Reply:
Def a WIP. I'm trying to move repetitive logic out of the game_over method and move some logic into report_game_result.

Reply:
👍